### ISSUE TITLE
[FR] Show if code is always possible / never possible to be executed with return types

### ISSUE THREAD

**Issue by andreasschroth** [kalessil/phpinspectionsea#1169]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Feature Request                          |
| Plugin         | Php Inspections (EA Ultimate), 2.0.11 |
| Language level | PHP 7.3                        |

Current behaviour:
---

Here is some example code:

```php
<?php

declare(strict_types=1);

function getInt(): int
{
    return 1;
}

function getBool(): bool
{
    return true;
}

if (getInt() !== getBool()) {
    echo 'Always executed!';
}
```

No warning is displayed.

Suggested behaviour:
---

Obviously the above code will always be executed as the returned integer will never be a boolean with strict comparison, no matter what integer or boolean is ever be returned. So I suggest that a warning is displayed for such cases.

**Comment by kalessil** [comment]
Since I still need to release 2.0.12, I'll include this =)

**Comment by andreasschroth** [comment]
@kalessil Thanks, awesome! That one would have saved us from a bug that even made it into production.

**Comment by kalessil** [comment]
Implemented!
